### DISCUSSION TITLE
Codespace running on IntelliJ does not respect postAttachCommand

### DISCUSSION THREAD

**Discussion by jmhon08** [community/community#64556] [Codespaces]
### Select Topic Area

Bug

### Body

I have this in my devcontainer.json

```
  "workspaceFolder": "/workspaces/ado-repos",
  "initializeCommand": "mkdir -p ${localWorkspaceFolder}/../ado-repos && curl -Lo- https://mdecodespaces.blob.core.windows.net/cf9308895e143/install-mde.sh | bash", 
  "onCreateCommand": "/workspaces/ruby-codespaces/setup.sh",
  "postAttachCommand": "/workspaces/ruby-codespaces/post-attach.sh",
```

but when I create and connect to my Codespace in IntelliJ, I notice the post-attach.sh script is never run. If I manually run the script in the Codespace terminal, the script runs fine. This issue does not happen in VS Code. If it helps to know, my dotfile is imported and runs as expected (so that part of the setup flow is fine).

**Comment by Kaysuto** [comment]
Hello,

It seems that running postAttachCommand in devcontainer.json does not work properly when using IntelliJ with GitHub Codespaces.

Some things to check:

- Make sure you are using the latest version of IntelliJ and the GitHub Codespaces plugin
- Verify the post-attach.sh script has execute permissions (chmod +x post-attach.sh)
- Try calling the script directly from your .bashrc or .zshrc instead of the postAttachCommand
- Open an issue on the IntelliJ Codespaces plugin GitHub repo to report the problem
- As a workaround until a fix is implemented, manually call the post-attach.sh script after connecting to the Codespace
- You can also use an editor like VS Code to initially create the Codespace, then open in IntelliJ afterwards

Let me know if you have any other questions!

**Comment by jmhon08** [comment]
Actually I was told by someone who works closely with JetBrains folks this:

"That's a known feature gap. postAttachCommand is only implemented in VS Code clients for now since it's executed on the client side in order to run the command in a true interactive terminal that can take input or be canceled, etc.    The rest of the lifecycle hooks are all executed as separate background processes on the remote side and are independent of the connecting client"

The issue is backlogged but not currently being worked on. So I will close this discussion
